src/zcbenchmarks.cpp

```diff
@@ -322,3 +326,82 @@ double benchmark_increment_note_witnesses(size_t nTxs)
     return timer_stop(tv_start);
 }
 
+// Fake the input of a given block
+class FakeCoinsViewDB : public CCoinsViewDB {
+    uint256 hash;
+    ZCIncrementalMerkleTree t;
+
+public:
+    FakeCoinsViewDB(std::string dbName, uint256& hash) : CCoinsViewDB(dbName, 100, false, false), hash(hash) {}
+
+    bool GetAnchorAt(const uint256 &rt, ZCIncrementalMerkleTree &tree) const {
+        if (rt == t.root()) {
+            tree = t;
+            return true;
+        }
+        return false;
+    }
+
+    bool GetNullifier(const uint256 &nf) const {
+        return false;
+    }
+
+    uint256 GetBestBlock() const {
+        return hash;
+    }
+
+    uint256 GetBestAnchor() const {
+        return t.root();
+    }
+
+    bool BatchWrite(CCoinsMap &mapCoins,
+                    const uint256 &hashBlock,
+                    const uint256 &hashAnchor,
+                    CAnchorsMap &mapAnchors,
+                    CNullifiersMap &mapNullifiers) {
+        return false;
+    }
```

Comment:
No; because `fJustCheck` is set to true, no write-effects are triggered. This was necessary because I couldn't mock out those globals, both because of the assumptions `ConnectBlock` is written with and because the benchmarks as-written run on a live (regtest) `zcashd`.